sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -600,6 +600,20 @@ namespace pi2ur {
 
 inline pi_result piTearDown(void *PluginParameter) {
   std::ignore = PluginParameter;
+  // Fetch the 1 known adapter so we can release it. Fetching it for a second
+  // time increases the reference count, so we need to release it twice.
+  static std::once_flag AdapterReleaseFlag;
+  ur_adapter_handle_t Adapter;
+  ur_result_t Ret = UR_RESULT_SUCCESS;
+  std::call_once(AdapterReleaseFlag, [&]() {
+    Ret = urAdapterGet(1, &Adapter, nullptr);
+    if (Ret == UR_RESULT_SUCCESS) {
+      Ret = urAdapterRelease(Adapter);
+      Ret = urAdapterRelease(Adapter);

```

Comment:
Double release, intentional?

Reply:
This is intentional as the reference count is known to be 2 by this point. The comment above should explain it, happy to clarify it if not though